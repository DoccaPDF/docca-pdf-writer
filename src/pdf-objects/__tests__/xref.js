import XRef from '../xref';
import Catalog from '../catalog';
import Trailer from '../trailer';

const xref = XRef({ startx: 105263,
  trailer: Trailer({
    ID: [ 'c7fbe2a37bce46eeaa1b16c7c84a7c08', 'c7fbe2a37bce46eeaa1b16c7c84a7c08' ],
    Size: 19,
    Info: {
      Title: 'My Title',
      Author: 'Me',
      Subject: 'Info Test',
      Keywords: 'test info',
      Creator: 'Jason',
      Producer: 'docca pdf writer',
      CreationDate: new Date('2017-02-13'),
      ModDate: new Date('2017-02-14'),
      Trapped: true,
      id: 18
    },
    Root: Catalog({ id: 2, Pages: { id: 3 } })
  }),
  offsets: {
    '1': 104443,
    '2': 104890,
    '3': 19
  }
});

describe('pdf-objects', () => {
  describe('xref', () => {
    it('creates a new object', () => {
      expect(xref).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(xref.toPDF()).toMatchSnapshot();
    });
  });
});
