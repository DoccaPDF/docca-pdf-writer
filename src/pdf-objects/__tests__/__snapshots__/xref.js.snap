// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects xref creates a new object 1`] = `
Object {
  "offsets": Object {
    "1": 104443,
    "2": 104890,
    "3": 19,
  },
  "startx": 105263,
  "trailer": Object {
    "ID": Array [
      "c7fbe2a37bce46eeaa1b16c7c84a7c08",
      "c7fbe2a37bce46eeaa1b16c7c84a7c08",
    ],
    "Info": Object {
      "Author": "Me",
      "CreationDate": 2017-02-13T00:00:00.000Z,
      "Creator": "Jason",
      "Keywords": "test info",
      "ModDate": 2017-02-14T00:00:00.000Z,
      "Producer": "docca pdf writer",
      "Subject": "Info Test",
      "Title": "My Title",
      "Trapped": true,
      "id": 18,
    },
    "Root": "2 0 R",
    "Size": 19,
  },
}
`;

exports[`pdf-objects xref returns a PDF object 1`] = `
"xref
0 4
0000000000 65535 f 
0000104443 00000 n 
0000104890 00000 n 
0000000019 00000 n 
trailer
<<
/ID [<c7fbe2a37bce46eeaa1b16c7c84a7c08><c7fbe2a37bce46eeaa1b16c7c84a7c08>]
/Size 19
/Info 18 0 R
/Root 2 0 R
>>

startxref
105263
%%EOF"
`;
